### PR TITLE
Login process integration

### PR DESCRIPTION
**Pull request by ex1st0r**

### CODE REVIEW

File: packages/login-ui/components/AccountSelector/AccountSelector.js [hunk 1]
```diff
@@ -3,86 +3,127 @@
...
 
-  handleChange = (event, index, value) => {
-    this.props.selectAccount(value)
+  renderWalletsList (){
+    const { onWalletSelect, walletsList } = this.props
+
+    if (!walletsList || !walletsList.length){
+      return (
+        <div styleName='empty-list'>
+          Sorry, there are no accounts to display
```
Comment (wAISw) [inline packages/login-ui/components/AccountSelector/AccountSelector.js]: Add Translate component

File: packages/login-ui/components/AccountSelector/AccountSelector.js [hunk 2]
```diff
@@ -3,86 +3,127 @@
...
-                size={24}
-                thickness={1.5}
-              /> : <Translate value='AccountSelector.selectAddress' />}
-              onClick={this.handleSelect}
-              disabled={!selectedAccount || isLoading}
-            />
+      <MuiThemeProvider>
+        <div styleName='wrapper'>
+
+          <div styleName='page-title'>My Accounts</div>
```
Comment (wAISw) [inline packages/login-ui/components/AccountSelector/AccountSelector.js]: Add Translate component

File: packages/login-ui/components/ConfirmMnemonic/ConfirmMnemonic.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+  render () {
+    const { handleSubmit, error } = this.props
+    return (
+      <MuiThemeProvider>
+        <form styleName='form' name={FORM_CONFIRM_MNEMONIC} onSubmit={handleSubmit}>
+          <div>
+            <div styleName='page-title'>Confirm back-up phrase</div>
+
+            <p styleName='description'>Click on back-up phrase words in the correct order.</p>
```
Comment (wAISw) [inline packages/login-ui/components/ConfirmMnemonic/ConfirmMnemonic.js]: Add Translate component

File: packages/login-ui/components/ConfirmMnemonic/ConfirmMnemonic.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+            </div>
+
+            <div styleName='controlsBlock'>
+              <div styleName='control' onClick={this.clearMnemonic.bind(this)}>Start Over</div>
+              <div styleName='control' onClick={this.clearLastWord.bind(this)}>Undo</div>
+            </div>
+
+            <div styleName='actions'>
+              <Button styleName='submit' type='submit' buttonType='login'>
+                Done
```
Comment (wAISw) [inline packages/login-ui/components/ConfirmMnemonic/ConfirmMnemonic.js]: Add Translate component

File: packages/login-ui/components/GenerateMnemonic/GenerateMnemonic.js [hunk 5]
```diff
@@ -3,80 +3,113 @@
...
+              <div styleName='passPhrase'>{ this.props.mnemonic }</div>
+              <div styleName='printButtonWrapper'>
+                <div styleName='printButton' onClick={() => {}}>
+                  <img src={PrintIcon} alt='' />
+                </div>
+              </div>
+            </div>
+
+            <div styleName='infoBlock'>
+              <div styleName='infoBlockHeader'>Important! Read the security guidelines</div>
```
Comment (wAISw) [inline packages/login-ui/components/GenerateMnemonic/GenerateMnemonic.js]: Add Translate component

File: packages/login-ui/components/GenerateMnemonic/GenerateMnemonic.js [hunk 6]
```diff
@@ -3,80 +3,113 @@
...
+
+              <ol styleName='infoBlockList'>
+                <li>
+                  <p styleName='listItemContent'>
+                    <b>Don&apos;t share your back-up phrase (mnemonic key) with someone you don&apos;t trust.</b>
```
Comment (wAISw) [inline packages/login-ui/components/GenerateMnemonic/GenerateMnemonic.js]: Add Translate component

File: packages/login-ui/components/GenerateMnemonic/GenerateMnemonic.js [hunk 7]
```diff
@@ -3,80 +3,113 @@
...
+                    &nbsp;Double check services you&apos;re giving your mnemonic to and don&apos;t share your phrase with anyone.
```
Comment (wAISw) [inline packages/login-ui/components/GenerateMnemonic/GenerateMnemonic.js]: Add Translate component

File: packages/login-ui/components/GenerateWallet/GenerateWallet.js [hunk 8]
```diff
@@ -3,131 +3,81 @@
...
-            ) : (
-              <div styleName='hint'><Translate value='GenerateWallet.walletSuccess' /></div>
-            )}
+      <MuiThemeProvider>
+        <div styleName='wrapper'>
+          <div>
+            <div styleName='page-title'>Download a Wallet File</div>
+
+            <p styleName='description'>
+              You can use this wallet file in password recovery option to
```
Comment (wAISw) [inline packages/login-ui/components/GenerateWallet/GenerateWallet.js]: Add Translate component

File: packages/login-ui/components/GenerateWallet/GenerateWallet.js [hunk 9]
```diff
@@ -3,131 +3,81 @@
...
-                />
-                <Warning />
-              </div>
-            ) : (
-              <div styleName='hint'><Translate value='GenerateWallet.walletSuccess' /></div>
-            )}
+      <MuiThemeProvider>
+        <div styleName='wrapper'>
+          <div>
+            <div styleName='page-title'>Download a Wallet File</div>
```
Comment (wAISw) [inline packages/login-ui/components/GenerateWallet/GenerateWallet.js]: Add Translate component

File: packages/login-ui/components/GenerateWallet/GenerateWallet.js [hunk 10]
```diff
@@ -3,131 +3,81 @@
...
-                label={<Translate value='GenerateWallet.continue' />}
-                disabled={!isDownloaded && (!isWarningSuppressed || !isPasswordValid)}
-                onClick={this.handleGenerateWalletClick}
-              />
+                styleName='button'
+                onClick={downloadWallet}
+              >
+                <img src={Wallet} alt='' />
+                <br />
+                Download
```
Comment (wAISw) [inline packages/login-ui/components/GenerateWallet/GenerateWallet.js]: Add Translate component

File: packages/login-ui/components/LoginWithPrivateKey/LoginWithPrivateKey.js [hunk 11]
```diff
@@ -3,86 +3,110 @@
...
-                  style={{ verticalAlign: 'middle', marginTop: -2 }}
-                  size={24}
-                  thickness={1.5}
-                /> : <Translate value='LoginWithPrivateKey.loginWithPrivateKey' />}
-              disabled={!isValidated || isLoading}
-              onClick={() => this.props.onLogin(privateKey)}
+          <div styleName='page-title'>Private key form</div>
+
+          <div styleName='description'>
+            Type or copy your private key into the box below
```
Comment (wAISw) [inline packages/login-ui/components/LoginWithPrivateKey/LoginWithPrivateKey.js]: Add Translate component

File: packages/login-ui/components/RecoverAccount/RecoverAccount.js [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+  render () {
+    const { handleSubmit, selectedWallet, error } = this.props
+
+    const wordsArray = new Array(12).fill()
+
+    return (
+      <MuiThemeProvider>
+        <form styleName='form' name={FORM_RECOVER_ACCOUNT} onSubmit={handleSubmit}>
+          <div styleName='title'>
+            Enter mnemonic to reset password
```
Comment (wAISw) [inline packages/login-ui/components/RecoverAccount/RecoverAccount.js]: Add Translate component

File: package.json [hunk 13]
```diff
@@ -129,7 +129,11 @@
     "truffle": "^4.1.11",
     "truffle-contract": "^3.0.5",
     "uglify-es": "github:mishoo/UglifyJS2#harmony",
-    "web3": "^0.20.1"
+    "uport-connect": "^0.6.0",
```
Comment (ozalexo) [inline package.json]: Alexander, could you please have a look into PR https://github.com/ChronoBank/ChronoMint/pull/691/files ?

Now we have multiple package.json files:
<root>/package.json
<root>/packages/core/package.json
<root>/packages/login/package.json
<root>/packages/login-ui/package.json

Added dependencies int his PR are moved into @chronobank/core and @chronobank/login pacakges.
Could you revert your changes in this file?

File: packages/login-ui/components/GenerateWallet/lang.js [hunk 14]
```diff
@@ -4,10 +4,10 @@
...
-  enterPassword: 'Enter password for the new wallet:',
-  password: 'Password',
-  passwordWarning: 'At least 8 characters',
-  iUnderstand: 'I\u00a0understand',
-  continue: 'Download Wallet',
-  walletSuccess: 'Your wallet has been generated',
+  title: 'Download a Wallet File',
+  description: 'You can use this wallet file in password recovery option to\n' +
+  '              make your account available in another browser, for example.',
+  descriptionExtra: 'The file is protected by the same password as your created before.',
```
Comment (ozalexo) [inline packages/login-ui/components/GenerateWallet/lang.js]: Typo: The file is protected by the same password as you[r] created before

your -> you

File: packages/login-ui/components/LoginWithMnemonic/LoginWithMnemonic.js [hunk 15]
```diff
@@ -3,118 +3,117 @@
...
+import styles from 'layouts/Splash/styles'
+import { Button } from 'components'
+import {
+  onSubmitMnemonicLoginForm,
+  onSubmitMnemonicLoginFormSuccess,
+  onSubmitMnemonicLoginFormFail,
+} from '@chronobank/login/redux/network/actions'
 
 import './LoginWithMnemonic.scss'
+import bip39 from "bip39";
```
Comment (ozalexo) [inline packages/login-ui/components/LoginWithMnemonic/LoginWithMnemonic.js]: [Minor]
Please replace " to ' (double quotes to single quotes)

File: packages/login-ui/components/NetworkSelector/NetworkSelector.js [hunk 16]
```diff
@@ -57,6 +57,7 @@ export default class NetworkSelector extends PureComponent {
 
   resolveNetwork = () => {
     const web3 = new Web3()
+    console.log('resolve network', this.props.getProviderURL())
```
Comment (ozalexo) [inline packages/login-ui/components/NetworkSelector/NetworkSelector.js]: Please remove console.log.
Or add //eslint-disable-next-line no-consolle if console.log really required.

File: packages/login/network/settings.js [hunk 17]
```diff
@@ -268,6 +268,30 @@ export const getNetworksByProvider = (providerId, withLocal = false) => {
...
+  let networks = []
+
+  Object.keys(providerMap)
+    .filter((key) => !providerMap[key].disabled)
+    .forEach((key) => {
+      const provider = providerMap[key]
+
+      const networksProvider = getNetworksByProvider(provider && provider.id, withLocal)
+        .map((network) => ({
+          provider: provider,
```
Comment (ozalexo) [inline packages/login/network/settings.js]: [MINOR]

This line also may be replaced to `provider,` as `network` below

File: src/layouts/partials/PersistWrapper/PersistWrapper.js [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+
+  constructor (props, context) {
+    super(props, context)
+    this.store = context.store
+  }
+
+  renderLoader(){
+    return (
+      <div styleName='loadingMessage'>
+        {/*<div styleName='loadingMessageHeader'>Log In</div>*/}
```
Comment (ozalexo) [inline src/layouts/partials/PersistWrapper/PersistWrapper.js]: [MINOR]
Is it still required or can be deleted?

File: src/layouts/partials/SidePanel/SidePanel.jsx [hunk 19]
```diff
@@ -5,7 +5,7 @@
 
 import { connect } from "react-redux"
 import PropTypes from "prop-types"
-import React, { PureComponent } from 'react'
+import React, { PureComponent, cloneElement } from 'react'
```
Comment (ozalexo) [inline src/layouts/partials/SidePanel/SidePanel.jsx]: cloneElement is now used. Please delete

File: src/redux/configureStore.js [hunk 20]
```diff
@@ -84,11 +84,14 @@ const configureStore = () => {
       const i18nState = state.get('i18n')
       const mainWalletsState = state.get('mainWallet')
       const walletsState = state.get('multisigWallet')
+      const persistAccount = state.get('persistAccount')
+      console.log('persistAccount', persistAccount)
```
Comment (ozalexo) [inline src/redux/configureStore.js]: Please delete console.log

**Pull request merged**